router/router.go

```diff
@@ -1023,19 +1015,12 @@ func (worker *workerT) postStatusOnResponseQ(respStatusCode int, payload json.Ra
 			if respStatusCode != types.RouterTimedOutStatusCode && respStatusCode != types.RouterUnMarshalErrorCode {
 				if timeElapsed > worker.rt.retryTimeWindow && status.AttemptNum >= worker.rt.maxFailedCountForJob {
 					status.JobState = jobsdb.Aborted.State
-					worker.retryForJobMapMutex.Lock()
-					delete(worker.retryForJobMap, destinationJobMetadata.JobID)
-					worker.retryForJobMapMutex.Unlock()
 				} else {
-					worker.retryForJobMapMutex.Lock()
-					worker.retryForJobMap[destinationJobMetadata.JobID] = time.Now().Add(durationBeforeNextAttempt(status.AttemptNum))
-					worker.retryForJobMapMutex.Unlock()
+					status.RetryTime = time.Now().Add(durationBeforeNextAttempt(status.AttemptNum))
 				}
 			}
 		} else if respStatusCode == 429 {
-			worker.retryForJobMapMutex.Lock()
-			worker.retryForJobMap[destinationJobMetadata.JobID] = time.Now().Add(durationBeforeNextAttempt(status.AttemptNum))
-			worker.retryForJobMapMutex.Unlock()
+			status.RetryTime = time.Now().Add(durationBeforeNextAttempt(status.AttemptNum))
```

Comment:
can do this once for both cases(500 fail, 429) a few lines below after the `jobState` has been decided.

```
if status.JobState == jobsdb.Failed.State {

    status.RetryTime = time.Now().Add(durationBeforeNextAttempt(status.AttemptNum))

}
```

Reply:
Unfortunately `jobState` can be amended later to `aborted`. Will try to refactor 👍

Reply:
LGTM
Only complaint is things go way beyond 80 columns in some places :-p

Reply:
Have you ever used a [Punch Card](https://blogs.sap.com/2022/07/26/the-magic-80-char-limit-which-used-to-exist-for-developers/)?!